specs/core/0_beacon-chain.md

```diff
@@ -742,17 +753,23 @@ def get_randao_mix(state: BeaconState,
     return state.randao_mixes[epoch % EPOCHS_PER_HISTORICAL_VECTOR]
 ```
 
-### `get_active_index_root`
+### `get_compact_committee_root`
 
 ```python
-def get_active_index_root(state: BeaconState,
-                          epoch: Epoch) -> Hash:
+def get_compact_committee_root(state: BeaconState, epoch: Epoch) -> Hash:
     """
-    Return the index root at a recent ``epoch``.
-    ``epoch`` expected to be between
-    (current_epoch - EPOCHS_PER_HISTORICAL_VECTOR + ACTIVATION_EXIT_DELAY, current_epoch + ACTIVATION_EXIT_DELAY].
+    Return the compact committee root for the current epoch.
     """
-    return state.active_index_roots[epoch % EPOCHS_PER_HISTORICAL_VECTOR]
+    committee_data = CompactCommitteees().data
+    for committee_number in range(get_epoch_committee_count(state, epoch)):
+        shard = (get_epoch_start_shard(state, epoch) + committee_number) % SHARD_COUNT
+        for index in get_crosslink_committee(state, epoch, shard):
+            validator = validators[index]
+            committee_data[shard].pubkeys.append(validator.pubkey)
+            # `index` (top 7 bytes) + `slashed` (8th bit) + `effective_balance` (bottom 7 bits)
+            compact_validator = index << 8 + validator.slashed << 7 + validator.effective_balance // GWEI_PER_ETH
```

Comment:
Does this break if validator balance `> 127`? (`2**7`)
Suggest we allocate more bits here

Reply:
The effective balance is capped at 32 ETH.

Reply:
Right

My main concern with this section is how specifically it is tuned to our section of constants

Reply:
Made the `compact_balance` 15 bits :)